packages/nx/src/plugins/js/index.ts

```diff
@@ -1,11 +1,88 @@
 import { ProjectGraphProcessor } from '../../config/project-graph';
 import { ProjectGraphBuilder } from '../../project-graph/project-graph-builder';
 import { buildNpmPackageNodes } from './project-graph/build-nodes/build-npm-package-nodes';
+import { buildExplicitDependencies } from './project-graph/build-dependencies/build-dependencies';
+import { readNxJson } from '../../config/configuration';
+import { fileExists, readJsonFile } from '../../utils/fileutils';
+import { PackageJson } from '../../utils/package-json';
+import {
+  lockFileExists,
+  lockFileHash,
+  parseLockFile,
+} from './lock-file/lock-file';
+import { NrwlJsPluginConfig, NxJsonConfiguration } from '../../config/nx-json';
+import { dirname, join } from 'path';
+import { projectGraphCacheDirectory } from '../../utils/cache-directory';
+import { readFileSync, writeFileSync } from 'fs';
+import { workspaceRoot } from '../../utils/workspace-root';
+import { ensureDirSync } from 'fs-extra';
+import { removeNpmNodes } from 'nx/src/plugins/js/lock-file/remove-npm-nodes';
 
-export const processProjectGraph: ProjectGraphProcessor = (graph) => {
+export const processProjectGraph: ProjectGraphProcessor = (graph, context) => {
   const builder = new ProjectGraphBuilder(graph);
 
+  const packageJson = readJsonFile<PackageJson>(
+    join(workspaceRoot, 'package.json')
+  );
```

Comment:
This will throw in `.nx` / encapsulated setups - Can we check if it exists first?